code/modules/chemistry/tools/dispensers.dm

```diff
@@ -397,6 +397,53 @@
 		..()
 		reagents.add_reagent("beer",1000)
 
+/obj/reagent_dispensers/chemicalbarrel
+	name = "chemical barrel"
+	desc = "For storing medical chemicals and less savory things. It can be labeled with a pen."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "barrel"
+	amount_per_transfer_from_this = 25
+	p_class = 3
+	flags = FPRINT | FLUID_SUBMERGE | OPENCONTAINER
+	var/labeled = FALSE
+
+	attackby(obj/item/W, mob/user)
+		if (istype(W, /obj/item/pen) && !src.labeled)
+			var/t = input(user, "Enter label", "Label", src.name) as null|text
+			if(t && t != src.name)
+				phrase_log.log_phrase("barrel", t, no_duplicates=TRUE)
+			t = copytext(strip_html(t), 1, 24)
+			if (isnull(t) || !length(t) || t == " ")
+				return
+			if (!in_interact_range(src, user) && src.loc != user)
+				return
+
+			src.name = t
+			src.labeled = TRUE
+			src.desc = "For storing medical chemicals and less savory things."
+		else
+			..()
+			return
+
+	bullet_act()
+		src.reagents.temperature_reagents(4000, 400) //exactly how a igniter works, expect to need a couple shots to make whatever is inside
+		src.reagents.temperature_reagents(4000, 400) //catch fire
+		playsound(src.loc, 'sound/impact_sounds/Metal_Hit_Heavy_1.ogg', 30, 1)
```

Comment:
😕 
I'd rather it just cause it to leak?  But for the sake of video games vibes which I'm assuming this is to elicit, this should really factor in aspects of the projectile as a foam dart shouldn't cause this behavior.

Reply:
Yeah this doesn't make sense to me. Why would bullets heat up a barrel?